# Initial Basic EDA

***This section serves as a preliminary step before the main exploratory analysis. The goal is to perform light but essential cleaning, specifically targeting missing values and duplicates to lay the groundwork for a deeper, more insightful EDA. That deeper analysis will then close with a comprehensive cleaning phase, leaving the dataset in a tidy, readable state.***

In [1]:
import pandas as pd
from pathlib import Path

script_path = Path.cwd()
data_path = script_path.parent/"data"/"raw"/"games.csv"

df = pd.read_csv(data_path)
df.shape
df.head(5)

,AppID,Name,Release date,Estimated owners,Peak CCU,Required age,Price,DiscountDLC count,About the game,Supported languages,...,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Developers,Publishers,Categories,Genres,Tags,Screenshots,Movies
2539430,Black Dragon Mage Playtest,"Aug 1, 2023",0 - 0,0,0,0.00,0,0,NaN,[],...,0,0,0,NaN,NaN,NaN,NaN,NaN,https://shared.akamai.steamstatic.com/store_it...,NaN
496350,Supipara - Chapter 1 Spring Has Come!,"Jul 29, 2016",0 - 20000,0,0,5.24,65,0,"Springtime, April: when the cherry trees come ...",['English'],...,0,8,0,minori,MangaGamer,"Single-player,Steam Trading Cards,Steam Cloud,...",Adventure,"Adventure,Visual Novel,Anime,Cute",https://shared.akamai.steamstatic.com/store_it...,NaN
1034400,Mystery Solitaire The Black Raven,"May 6, 2019",0 - 20000,0,0,4.99,0,0,"Immerse yourself in the most beloved, mystical...","['English', 'French', 'German', 'Russian']",...,0,0,0,Somer Games,8floor,"Single-player,Family Sharing",Casual,"Casual,Card Game,Solitaire,Puzzle,Hidden Objec...",https://shared.akamai.steamstatic.com/store_it...,NaN
3292190,버튜버 파라노이아 - Vtuber Paranoia,"Oct 31, 2024",0 - 20000,1,0,8.99,0,1,"synopsis 'Hello, I'm Hiyoro, a new YouTuber!' ...",['Korean'],...,0,0,0,유진게임즈,유진게임즈,"Single-player,Steam Achievements,Family Sharing","Casual,Indie,Simulation",NaN,https://shared.akamai.steamstatic.com/store_it...,NaN
3631080,Maze Quest VR,"Apr 24, 2025",0 - 20000,0,0,4.99,0,0,Its not just a Maze; its a Quest! Enter the ca...,['English'],...,0,0,0,Reality Expanded LLC,Reality Expanded LLC,"Single-player,VR Only,Steam Leaderboards,Famil...","Action,Early Access",NaN,https://shared.akamai.steamstatic.com/store_it...,NaN


***From this initial graph, we can immediately spot a data loading issue: the AppID column appears misaligned with its values. Since the raw CSV file shows no such problem, this is a pandas specific artifact, pandas is incorrectly treating one of the columns as the row index, shifting everything out of place.***

***The fix is straightforward, explicitly passing index_col=False when reading the file forces pandas to use its own default integer index, restoring the correct column alignment.***

In [2]:
df = pd.read_csv(data_path, index_col=False)
df.head(5)

,AppID,Name,Release date,Estimated owners,Peak CCU,Required age,Price,DiscountDLC count,About the game,Supported languages,...,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Developers,Publishers,Categories,Genres,Tags,Screenshots,Movies
0,2539430,Black Dragon Mage Playtest,"Aug 1, 2023",0 - 0,0,0,0.00,0,0,NaN,...,0,0,0,0,NaN,NaN,NaN,NaN,NaN,https://shared.akamai.steamstatic.com/store_it...
1,496350,Supipara - Chapter 1 Spring Has Come!,"Jul 29, 2016",0 - 20000,0,0,5.24,65,0,"Springtime, April: when the cherry trees come ...",...,8,0,8,0,minori,MangaGamer,"Single-player,Steam Trading Cards,Steam Cloud,...",Adventure,"Adventure,Visual Novel,Anime,Cute",https://shared.akamai.steamstatic.com/store_it...
2,1034400,Mystery Solitaire The Black Raven,"May 6, 2019",0 - 20000,0,0,4.99,0,0,"Immerse yourself in the most beloved, mystical...",...,0,0,0,0,Somer Games,8floor,"Single-player,Family Sharing",Casual,"Casual,Card Game,Solitaire,Puzzle,Hidden Objec...",https://shared.akamai.steamstatic.com/store_it...
3,3292190,버튜버 파라노이아 - Vtuber Paranoia,"Oct 31, 2024",0 - 20000,1,0,8.99,0,1,"synopsis 'Hello, I'm Hiyoro, a new YouTuber!' ...",...,0,0,0,0,유진게임즈,유진게임즈,"Single-player,Steam Achievements,Family Sharing","Casual,Indie,Simulation",NaN,https://shared.akamai.steamstatic.com/store_it...
4,3631080,Maze Quest VR,"Apr 24, 2025",0 - 20000,0,0,4.99,0,0,Its not just a Maze; its a Quest! Enter the ca...,...,0,0,0,0,Reality Expanded LLC,Reality Expanded LLC,"Single-player,VR Only,Steam Leaderboards,Famil...","Action,Early Access",NaN,https://shared.akamai.steamstatic.com/store_it...


In [3]:
df.info()

null_counts = df.isnull().sum()
null_pct = ((null_counts / len(df)) * 100).round(2)
null_summary = pd.DataFrame({"null_count": null_counts, "null_pct": null_pct})
null_summary[null_summary["null_count"] > 0].sort_values(by="null_count", ascending=False)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 122611 entries, 0 to 122610
Data columns (total 39 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   AppID                       122611 non-null  int64  
 1   Name                        122610 non-null  object 
 2   Release date                122611 non-null  object 
 3   Estimated owners            122611 non-null  object 
 4   Peak CCU                    122611 non-null  int64  
 5   Required age                122611 non-null  int64  
 6   Price                       122611 non-null  float64
 7   DiscountDLC count           122611 non-null  int64  
 8   About the game              122611 non-null  int64  
 9   Supported languages         114162 non-null  object 
 10  Full audio languages        122611 non-null  object 
 11  Reviews                     122611 non-null  object 
 12  Header image                12070 non-null   object 
 13  Website       

,null_count,null_pct
Achievements,122571,99.97
User score,118355,96.53
Header image,110541,90.16
Average playtime forever,100153,81.68
Support url,72935,59.48
Support email,68469,55.84
Screenshots,39265,32.02
Windows,22263,18.16
Genres,8953,7.30
Categories,8909,7.27


***Using `df.info()`, we can see that this dataset contains **122,611 entries across 39 columns**, where each entry represents a game (including duplicates, as the dataset has not yet been cleaned). These 39 columns cover a wide range of metrics describing and evaluating each game, making the dataset considerably larger and more complex than what this project requires.***

***Several columns are unlikely to contribute meaningful value to our analysis. Most notably, `Achievements` contains up to **99.97% null values**, making it nearly unusable in its current state. Other columns also exhibit high rates of nullness, as shown in the null value summary above.***

***Beyond missing data, a number of columns are stored in incorrect data types. For example, `Release date` is stored as `object` rather than `datetime`, and `Average playtime forever` is typed as `object` despite the three other playtime-related columns (`Average playtime two weeks`, `Median playtime forever`, `Median playtime two weeks`) all being stored as `int64`. Additionally, several columns appear to have been assigned the wrong dtype entirely, for instance, `Metacritic score` is stored as `bool` and `Metacritic url` as `int64`, which is clearly inconsistent with the kind of data these fields should hold.***

***All of these issues: duplicate entries, high null columns, and dtype inconsistencies will be addressed in the initial cleaning step.***

In [4]:
print(f"Duplicate rows: {df.duplicated().sum()}")
print(f"Duplicate AppIDs: {df['AppID'].duplicated().sum()}")

Duplicate rows: 0
Duplicate AppIDs: 0


In [5]:
df.describe()

,AppID,Peak CCU,Required age,Price,DiscountDLC count,About the game,Metacritic url,Positive,Negative,Score rank,Achievements,Recommendations,Notes,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Developers
count,1.226110e+05,1.226110e+05,122611.000000,122611.000000,122611.000000,122611.000000,122611.000000,122611.000000,1.226110e+05,1.226110e+05,40.000000,122611.000000,1.226110e+05,1.226110e+05,122611.000000,1.226110e+05,122611.000000
mean,1.985386e+06,5.459332e+01,0.167611,4.765091,18.353663,0.545856,2.564941,0.024549,1.044986e+03,1.691974e+02,99.175000,18.087015,9.618250e+02,2.080232e+02,13.789268,1.735705e+02,14.722170
std,1.087595e+06,3.729452e+03,1.653591,12.531030,28.858970,14.516026,13.660559,1.394901,2.809173e+04,5.374645e+03,0.675107,141.493879,2.187880e+04,1.121768e+04,270.378053,1.120254e+04,294.509615
min,1.000000e+01,0.000000e+00,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000e+00,0.000000e+00,98.000000,0.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000
25%,1.063175e+06,0.000000e+00,0.000000,0.550000,0.000000,0.000000,0.000000,0.000000,0.000000e+00,0.000000e+00,99.000000,0.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000
50%,1.907380e+06,0.000000e+00,0.000000,2.240000,0.000000,0.000000,0.000000,0.000000,5.000000e+00,1.000000e+00,99.000000,2.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000
75%,2.869560e+06,0.000000e+00,0.000000,5.240000,40.000000,0.000000,0.000000,0.000000,3.700000e+01,1.000000e+01,100.000000,19.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000
max,4.264350e+06,1.013936e+06,21.000000,999.980000,100.000000,3703.000000,97.000000,100.000000,7.642084e+06,1.173003e+06,100.000000,9821.000000,4.830455e+06,3.429544e+06,20088.000000,3.429544e+06,20088.000000


***Before examining the statistics, duplicate checking confirmed that the dataset contains no repeated rows or duplicate `AppID` values, meaning each entry uniquely identifies a game.***

***`df.describe()` then reveals several dtype issues flagged earlier. Columns such as `Metacritic url`, `About the game`, `Developers`, `Notes`, and `Score rank` appear in the numeric summary despite clearly not representing quantitative values, further confirming they were assigned incorrect dtypes during ingestion.***

***Beyond dtype issues, the distributions reveal that the dataset is heavily skewed toward low engagement titles. The median values for `Peak CCU`, `Positive` reviews, `Negative` reviews, `Recommendations`, and all playtime related columns are all **0**, meaning more than half the games in the dataset have essentially no recorded player activity. This is worth keeping in mind during analysis, as a small number of high engagement outliers, such as the game with a Peak CCU of over 1,000,000 will heavily influence any mean based metrics.***

***`Price` ranges from free ($0) to $999.98, with a median of just $2.24, reflecting the prevalence of very cheap or free to play titles on the platform. Similarly, `Required age` has a mean of 0.17 and a maximum of 21, indicating that the vast majority of games carry no age restriction.***